### PR TITLE
mgr/telemetry: modify stats_per_pool and add stats_per_pg

### PR DESCRIPTION
**Pull request by ljflores**
We decided that it might be more useful to report data on a pg level in addition to what we were reporting for pool level. In the process of adding `stats_per_pg`, I realized that there was a much easier way to collect `stats_per_pool` than the current implementation. Fetching `pg_dump` from the mgr module already provides a field called `pool_stats` that is the same as aggregated pg stats.

All in all, this solution should provide the information we want, with a much cleaner implementation.

Signed-off-by: Laura Flores <lflores@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/telemetry/module.py [hunk 1]
```diff
@@ -885,7 +839,10 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
 
         if 'perf' in channels:
             report['perf_counters'] = self.gather_perf_counters()
-            report['stat_sum_per_pool'] = self.get_stat_sum_per_pool()
+
+            report['stats_per_pool'] = self.get('pg_dump')['pool_stats']
+            report['stats_per_pg'] = self.get('pg_dump')['pg_stats']
```
Comment (yaarith) [inline src/pybind/mgr/telemetry/module.py:844]: I would separate these changes into 2 commits: one with changes related only to `stats_per_pool`, and another with the new `report['stats_per_pg']` metrics collection.

**Review by yaarith — COMMENTED** [review]
looks good; I had one comment about separating commits.

**Comment by yaarith** [comment]
Clarifying for other reviewers:

* The new `report['stats_per_pool']` collection has everything `report['stat_sum_per_pool']` collected per pool, as well as the metrics: 
"acting", "log_size", "num_pg", "num_store_stats", "ondisk_log_size", "store_stats", and "up".

* Also, there are **no** pools names collected in:
```python
report['stats_per_pool'] = self.get('pg_dump')['pool_stats']
report['stats_per_pg'] = self.get('pg_dump')['pg_stats']
```

**Comment by ljflores** [comment]
Apologies for the delay @yaarith-- this should be good now.

**Review by yaarith — APPROVED** [review]
looks good;  let's remove the the backport message from the second commit since it's relevant only to the first one.

**Comment by ljflores** [comment]
> looks good; let's remove the the backport message from the second commit since it's relevant only to the first one.

Done!

**Comment by ljflores** [comment]
@yaarith should be properly rebased now.

**Comment by yaarith** [comment]
we need to remove the"Conflicts" section from the first commit message (70e24f8db2d7b8f9b1d033e14102b99405702a02).

I guess the last sentence in the commit message relates to the "Conflicts" part:
`Removed "stats_sum_per_pool" and restored the mode option in get_osd_histograms()`
it should also be removed since there were no changes related to get_osd_histograms() in this commit.

**Comment by ljflores** [comment]
> we need to remove the"Conflicts" section from the first commit message ([70e24f8](https://github.com/ceph/ceph/commit/70e24f8db2d7b8f9b1d033e14102b99405702a02)).
> 
> I guess the last sentence in the commit message relates to the "Conflicts" part: `Removed "stats_sum_per_pool" and restored the mode option in get_osd_histograms()` it should also be removed since there were no changes related to get_osd_histograms() in this commit.

I'll remove it. I thought that any time a commit is rebased, you need to include a "Conflicts" section. But I can see why it's not relevant here.

**Comment by yaarith** [comment]
I'm not sure if we should include the "Conflicts" section in case we rebase and squash commits; sorry for the trouble if we should!
@neha-ojha, should we include a "Conflicts" section in such cases?
I was confused by the message after that section, since there were no changes related to `get_osd_histograms`, and thought it had to do with a backport conflict, rather than the rebase one.

**Comment by ljflores** [comment]
> I'm not sure if we should include the "Conflicts" section in case we rebase and squash commits; sorry for the trouble if we should! @neha-ojha, should we include a "Conflicts" section in such cases? I was confused by the message after that section, since there were no changes related to `get_osd_histograms`, and thought it had to do with a backport conflict, rather than the rebase one.

Yeah this is a bit tricky. Essentially, conflicts occurred because the "mempool" and "histograms / perf counter" PRs were merged. Upon rebasing, I had to add in the newly-merged mempool stats, and include the newly-merged mode option for histograms. 

I know that including a "Conflicts" section is important in backports, but I'm not sure if it's needed here.

**Comment by neha-ojha** [comment]
> > I'm not sure if we should include the "Conflicts" section in case we rebase and squash commits; sorry for the trouble if we should! @neha-ojha, should we include a "Conflicts" section in such cases? I was confused by the message after that section, since there were no changes related to `get_osd_histograms`, and thought it had to do with a backport conflict, rather than the rebase one.
> 
> Yeah this is a bit tricky. Essentially, conflicts occurred because the "mempool" and "histograms / perf counter" PRs were merged. Upon rebasing, I had to add in the newly-merged mempool stats, and include the newly-merged mode option for histograms.
> 
> I know that including a "Conflicts" section is important in backports, but I'm not sure if it's needed here.
> 
> And there's no trouble! If it's needed, I can easily add it back in.

We should never have a "Conflicts" section in master commits. We should you rebase to cleanup/restructure such commits.

**Comment by ljflores** [comment]
jenkins test make check

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/lflores-2021-11-08_21:48:32-rados-wip-pg-stats-distro-default-smithi/ 
http://pulpito.front.sepia.ceph.com/lflores-2021-11-08_19:58:04-rados-wip-pg-stats-distro-default-smithi/ 

Failures unrelated, tracked in:

    [6492777, 6492791, 6491519, 6491534] -- seen in master: http://pulpito.front.sepia.ceph.com/yuriw-2021-11-06_17:01:58-rados-wip-yuri-testing-master-11.5.21-distro-basic-smithi/6488501/
    [6492779, 6492789, 6491522] -- tracked by https://tracker.ceph.com/issues/53206; seen in in master run: http://pulpito.front.sepia.ceph.com/yuriw-2021-11-06_17:01:58-rados-wip-yuri-testing-master-11.5.21-distro-basic-smithi/6488712/
    [6492785, 6492798, 6491527, 6491540] -- seen in master run: http://pulpito.front.sepia.ceph.com/yuriw-2021-11-06_17:01:58-rados-wip-yuri-testing-master-11.5.21-distro-basic-smithi/6488803/


Details:

    Bug_#53206: Found coredumps on ubuntu@smithi115.front.sepia.ceph.com | IndexError: list index out of range - Tools - Teuthology

**Review by yaarith — APPROVED** [review]

**Pull request merged**